subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -288,6 +289,22 @@ public void applyTo(boolean value, StartParameterInternal settings, Origin origi
         }
     }
 
+    public static class WatchFileSystemOption extends BooleanBuildOption<StartParameterInternal> {
+        public static final String LONG_OPTION = "watch-fs";
+
+        public WatchFileSystemOption() {
+            super("org.gradle.unsafe.watch-fs", BooleanCommandLineOptionConfiguration.create(
+                LONG_OPTION,
+                "Enables watching the file system for changes so that in-memory state of the file system can be re-used for the next build.",
```

Comment:
I think we should try to make this a bit less technical. Maybe:

```suggestion
                "Enables watching the file system for changes allowing data about the file system to be re-used for the next build.",
```